cpp/src/aztec3/circuits/abis/contract_deployment_data.hpp

```diff
@@ -18,10 +18,10 @@ template <typename NCT> struct ContractDeploymentData {
     typedef typename NCT::fr fr;
     typedef typename NCT::boolean boolean;
 
-    fr constructor_vk_hash;
-    fr function_tree_root;
-    fr contract_address_salt;
-    address portal_contract_address;
+    fr constructor_vk_hash = 0;
```

Comment:
Usually the base type's default constructor is zero-initialized so you don't need to do this everywhere, but perhaps some reason against it

Reply:
We need to because the the field class explicitly defines the default constructor to do nothing....

Reply:
That does not seem like a rational choice

Reply:
The body could easily be a memset(this, 0, sizeof(*this)) and the C++ compiler would now very well when to get rid of it